packages/router/test/router_scroller.spec.ts

```diff
@@ -291,3 +291,163 @@ function createRouterScroller({
 function setScroll(viewportScroller: any, x: number, y: number) {
   viewportScroller.getScrollPosition.and.returnValue([x, y]);
 }
+
+describe('withInMemoryScrolling', () => {
+  @Component({template: '<div>simple</div>'})
+  class SimpleComponent {}
+
+  function nextScrollEvent(events: Subject<Event | PrivateRouterEvents>): Promise<Scroll> {
+    return events
+      .pipe(
+        filter((e): e is Scroll => e instanceof Scroll),
+        take(1),
+      )
+      .toPromise() as Promise<Scroll>;
+  }
+
+  function createRouterWithInMemoryScrolling({
+    scrollPositionRestoration,
+    anchorScrolling,
+  }: InMemoryScrollingOptions) {
+    const viewportScroller = jasmine.createSpyObj('viewportScroller', [
+      'getScrollPosition',
+      'scrollToPosition',
+      'scrollToAnchor',
+      'setHistoryScrollRestoration',
+    ]);
+
+    TestBed.configureTestingModule({
+      providers: [
+        provideRouter(
+          [
+            {path: 'a', component: SimpleComponent},
+            {path: 'b', component: SimpleComponent},
+          ],
+          withInMemoryScrolling({scrollPositionRestoration, anchorScrolling}),
+        ),
+        {provide: ViewportScroller, useValue: viewportScroller},
+      ],
+    });
+
+    const events = new Subject<Event | PrivateRouterEvents>();
+    (TestBed.inject(NavigationTransitions) as Writable<NavigationTransitions>).events = events;
+
+    const scroller = TestBed.inject(ROUTER_SCROLLER);
+    scroller.init();
+    return {events, viewportScroller};
+  }
+
+  function setScroll(viewportScroller: any, x: number, y: number) {
+    viewportScroller.getScrollPosition.and.returnValue([x, y]);
+  }
+
+  it('should configure scrollPositionRestoration  and anchorScrolling as disabled by default', () => {
+    TestBed.configureTestingModule({
+      providers: [provideRouter([], withInMemoryScrolling())],
+    });
+
+    const scroller = TestBed.inject(ROUTER_SCROLLER);
+    expect((scroller as any).options.scrollPositionRestoration).toBe('disabled');
+    expect((scroller as any).options.anchorScrolling).toBe('disabled');
+  });
```

Comment:
nit: No need to resort to any for private property
```suggestion
    const scroller = TestBed.inject(ROUTER_SCROLLER);
    expect(scroller['options'].scrollPositionRestoration).toBe('disabled');
    expect(scroller['options'].options.anchorScrolling).toBe('disabled');
  });
```

Reply:
Done